releasenotes/notes/dagcircuit-separable-circuits-142853e69f530a16.yaml

```diff
@@ -0,0 +1,13 @@
+---
+features:
+  - |
+    This feature introduces a new `DAGCircuit` class method,
+    `separable_circuits`, which returns a list of `DAGCircuit` objects,
+    one for each mutually disconnected set of qubits in the `DAGCircuit` object.
```

Comment:
Kevin: your crosslinks look fine.  Technically the `~` in each of the `` :class:`~.DAGCircuit` ``s is superfluous (but not in the `:meth:` one - it serves a truncation purpose there) but it's not harmful and doesn't change the output.

Reply:
@jakelishman , if I remove the unnecessary `~`, does the `.` before the class name remain?

Reply:
Yeah.  The rules are approximately like this:
- by default, the display text will be what's between the `` ` ` ``, but with a leading `~` and/or `.` stripped away.  This text is also used to lookup the object, and if there's no prefix, then the object needs to match something in the Sphinx catalogue, at first within the current (Sphinx) scope, then if there's no match it goes up a level, etc.
- a leading `~` means that the displayed text is changed, so that only the last component (as separated by dots) is actually displayed.  Doesn't affect the lookup.
- a leading `.` causes a _complete_ suffix search; Sphinx will find the object _anywhere_ in the catalogue that matches that suffix.  If there's more than one, then it raises an error.

Most of the time we want to use `.` for lookup.  If we need to more explicitly spell out how to find an object, then we often want to use `~` to truncate it to just the last component, because otherwise the prose is hard to read.  Users can hover over the link (or click it!) to get the full form anyway, or it's usually very clear from context.

Reply:
Looking closer, in this particular case the leading `.` in the class names will actually be unnecessary too, because the methods are in a subscope of `DAGCircuit`.  I almost always just keep them in though, if not only because it's easier than trying to remember the exact scoping rules Sphinx uses.  They're not precisely the same as Python's - they flow from the docs structure, not the package hierarchy; in Qiskit they're _mostly_ similar, but not entirely.